app/assets/javascripts/single_page/dynamic_table.js.erb

```diff
@@ -47,19 +48,36 @@ const handleSelect = (e) => {
 						data = data[0]?.id ? data : [];
 						const existingObjectsJSON = data.map(({ id, title }) => ({
 							id: id,
-							name: title
+							title: title
 						}));
-						const existingObjects = JSON.stringify(existingObjectsJSON);
-						const url = typeaheadUrl.replace("%LINKED%", c.linked_sample_type);
-						if (options.readonly) {
-							return existingObjectsJSON.map((e) => `<span class="badge">${e.name}</span>`).join(" ");
+						const existingOptions = existingObjectsJSON.map((e) => `<option selected="selected" value=${e.id}>${e.title}</option>`).join("");
+						if (options.readOnly) {
+							return existingObjectsJSON.map((e) => `<span class="badge">${e.title}</span>`).join(" ");
+
 						} else {
-							return objectInput.replace("%EXISTING%", existingObjects).replace("%URL%", url);
+							const url = typeaheadUrl.replace("_LINKED_", c.linked_sample_type);
+							const objectInputName = 'objectInput[' + existingObjectsJSON.map((e) => e.id).join('-') + ']';
```

Comment:
I noticed it as well but since I didn't have access to the row / column index from the `mRender` block and I didn't know how to solve it immediately. So I  postponed it and in the end forgot about it.
I'm not sure this is the best way but I combined the sample ID's with a UUID. That way there is still some logic in the name of the `objectInputs`.